plugins/org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/interpreter/SwitchConstantExpressionsInterpreter.xtend

```diff
@@ -50,7 +50,7 @@ class SwitchConstantExpressionsInterpreter extends AbstractConstantExpressionsIn
 					if (feature.constant) {
 						return feature.constantValue
 					}
-				} else if (feature.final) {
+				} else if (feature.final && feature.static) {
```

Comment:
I think final and static is not sufficient, e.g. a field must be a compile time constant (see isSetConstant and isConstant).